dataflow_test.py

```diff
@@ -523,3 +523,71 @@ def test_count_filter_counts_correctly():
 
     assert result.n_passed == len( odd_numbers)
     assert result.n_failed == len(even_numbers)
+
+
+@parametrize("component",
+             (df.map   (lambda x: x)    ,
+              df.filter(lambda x: x > 0),
+              df.sink  (print)          ,
+              df.branch(df.sink(print)) ,
+              df.pipe  (df.map(abs))    ))
+def test_string_to_pick_ignores_components(component):
+    assert component is df.string_to_pick(component)
```

Comment:
Two points:

1. I don't think that `string_to_pick` is part of the public API (so it should start with an underscore!). If that is the case, then it shouldn't be tested here, because these are documentation tests aimed at users.

2. As you well know, I love `parametrize`, but I'm reluctant to use it in user-facing documentation tests, because a complete layman should be able to read these and understand pretty much anything. `parametrize` will freak them out.